src/riscv/lib/src/pvm/common.rs

```diff
@@ -460,6 +483,19 @@ impl<'a, MC: MemoryConfig, BCC: BlockCacheConfig, B: Block<MC, ProofGen<Ref<'a,
     }
 }
 
+/// An [`InputRequest`] is what the PVM expects as input for a specific tick.
+pub enum InputRequest {
+    /// No input is required at the moment, normal execution can continue.
+    NoInputRequired,
+    /// The PVM is waiting for an initial inbox input, which is the message for level 0, index 0.
+    Initial,
+    /// The PVM is waiting for a message at a specific level and inbox index.
+    /// `FirstAfter(level, counter)` represents the message at `level`, and index `counter + 1`.
+    FirstAfter(u32, u64),
```

Comment:
Explicit field names make misuse harder and make it obvious what each integer means without reading the comment.

```suggestion
    FirstAfter { level: u32, counter: u64 },
```